release.py

```diff
@@ -0,0 +1,290 @@
+"""
+This script is used to build the zip/tar archives included in GitHub releases.
+
+It has been written to run on ubuntu GitHub CI runner machines.
+Running this script on windows will likely result in errors!
+"""
+
+import platform
+import sys
+
+if platform.system() != "Linux":
+    print("This script should only run on a machine with a Linux OS.")
+    sys.exit(7)
+
+import os
+import git
+import itertools
+import re
+import typing
+import shutil
+import zipfile
+import glob
+import subprocess
+import logging
+
+from pathlib import Path
+
+BUILD_DIR = Path("/tmp/build")
+RELEASE_DIR = Path("./release")
+
+
+def init_logging():
+
+    logging.basicConfig(level=os.environ.get("LOG_LEVEL", "INFO"))
+
+    l = logging.getLogger()
+    l.handlers = []
+    std_out = logging.StreamHandler(stream=sys.stdout)
+    std_err = logging.StreamHandler(stream=sys.stdout)
+    std_out.setLevel(logging.INFO)
+    std_out.addFilter(lambda record: record.levelno < logging.ERROR)
+    std_err.setLevel(logging.ERROR)
+    l.addHandler(std_out)
+    l.addHandler(std_err)
+
+    return l
+
+
+logger = init_logging()
+
+
+def check_git_bin_exists() -> bool:
```

Comment:
should be safe to assume git is on a GitHub CI worker but i was being defensive in case someone tries to run this without git installed.